### PR TITLE
Enable Coal fuel type for Coil:Heating:Fuel

### LINKED ISSUE #8110: Coal fuel type not available in Coil:Heating:Fuel
Issue overview
--------------
E+’s `Coil:Heating:Fuel` object doesn’t support `Coal` fuel type for some reason. But some other objects with fuel type options do. From the Energy+.idd:

OtherEquipment | Coil:Heating:Fuel | Boiler:HotWater | WaterHeater:Tank
-- | -- | -- | --
\key NaturalGas | \key NaturalGas | \key NaturalGas | \key NaturalGas
\key Propane | \key Propane | \key Propane | \key Propane
\key FuelOilNo1 | \key FuelOilNo1 | \key FuelOilNo1 | \key FuelOilNo1
\key FuelOilNo2 | \key FuelOilNo2 | \key FuelOilNo2 | \key FuelOilNo2
\key Diesel | \key Diesel | \key Diesel | \key Diesel
\key Gasoline | \key Gasoline | \key Gasoline | \key Gasoline
\key Coal |   | \key Coal | \key Coal
\key OtherFuel1 | \key OtherFuel1 | \key OtherFuel1 | \key OtherFuel1
\key OtherFuel2 | \key OtherFuel2 | \key OtherFuel2 | \key OtherFuel2

Is there any specific reason that coal is excluded in `Coil:Heating:Fuel` object? [source code](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/HeatingCoils.cc#L670-L677) and idd is limiting its use, while the [docs](https://github.com/NREL/EnergyPlus/blob/develop/doc/input-output-reference/src/overview/group-heating-and-cooling-coils.tex#L1244), is inconsistently specifying coal as a valid option. 
Can we add `Coal` as a valid option?

### Details
Some additional details for this issue (if relevant):
 - Platform (Windows, 10)
 - Version of EnergyPlus (9.3.0)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by yzhou601**
Pull request overview
---------------------
 - Fixes #8110 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by yzhou601** [comment]
Locally tested a file changing its `Coil:Heating:Fuel` from FuelOilNo2 fuel type to Coal fuel type. There's no numeric difference, Heating Coil Fuel Energy and Fuel: Facility are now showing up with different output variables/meters. [Test files](https://drive.google.com/drive/folders/1ZR82kYkb9Y_a1O4nszdgvNnk5rloQrOh?usp=sharing)

File: tst/EnergyPlus/unit/HeatingCoils.unit.cc [hunk 1]
```diff
@@ -103,6 +103,20 @@ TEST_F(EnergyPlusFixture, HeatingCoils_FuelTypeInputError)
...
+        {"Coil:Heating:Fuel,", "  Furnace Coil,            !- Name", "  ,    !- Availability Schedule Name", "  Coal,                 !- FuelType",
+         "  0.8,                     !- Gas Burner Efficiency", "  20000,                   !- Nominal Capacity {W}",
+         "  Heating Coil Air Inlet Node,  !- Air Inlet Node Name", "  Air Loop Outlet Node;    !- Air Outlet Node Name"});
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    ASSERT_NO_THROW(HeatingCoils::GetHeatingCoilInput(state));
+
+    EXPECT_EQ(HeatingCoils::HeatingCoil(1).FuelType_Num, DataGlobalConstants::iRT_Coal);
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HeatingCoils.unit.cc:118]: :+1:

File: src/EnergyPlus/HeatingCoils.cc [hunk 2]
```diff
@@ -669,7 +669,7 @@ namespace HeatingCoils {
             coil.FuelType_Num = AssignResourceTypeNum(Alphas(3));
             if (!(coil.FuelType_Num == iRT_Natural_Gas || coil.FuelType_Num == iRT_Propane || coil.FuelType_Num == iRT_Diesel ||
                   coil.FuelType_Num == iRT_Gasoline || coil.FuelType_Num == iRT_FuelOil_1 || coil.FuelType_Num == iRT_FuelOil_2 ||
-                  coil.FuelType_Num == iRT_OtherFuel1 || coil.FuelType_Num == iRT_OtherFuel2) ||
+                  coil.FuelType_Num == iRT_OtherFuel1 || coil.FuelType_Num == iRT_OtherFuel2 || coil.FuelType_Num == iRT_Coal) ||
```
Comment (Myoldmopar) [inline src/EnergyPlus/HeatingCoils.cc:672]: Makes sense.

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -52323,6 +52323,7 @@ Coil:Heating:Fuel,
        \key Propane
        \key Diesel
        \key Gasoline
+       \key Coal
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:52326]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
One CI diff is unrelated.  Testing locally.

**Comment by Myoldmopar** [comment]
Testing results
- Took the ZoneWSHP_wDOAS file and trimmed out all irrelevant output variables, ran fine
- Converted from gas to fuel, ran fine, but error file lists out missing gas output variables -- a good sign
- Converted from Heating Coil Gas Rate to Heating Coil Coal Rate worked to clean up error file
- In the output of the coal CSV file I then changed the header from Coal Rate back to Gas Rate and found the files to be binary equal

This seems all good to me.  I'm going to merge.

**Comment by rraustad** [comment]
I wish I would have seen this sooner, but doesn't matter because it's not related. The report variable should be Heating Coil Fuel Rate so that it is the same regardless of fuel type. Easier on GUIs.

**Comment by mjwitte** [comment]
@rraustad Well, we could debate the fuel-specific output variables over in #8134.

**Comment by Myoldmopar** [comment]
> @rraustad Well, we could debate the fuel-specific output variables over in #8134.

Perfect.  Yeah, sorry I'm moving a little quick trying to keep the PR list low, @rraustad.  I am happy to make the variable name change over in #8134.

**Pull request merged**